File: WorldFeed/Server/src/Services/Post/WorldFeed.Post.API/Helpers/IWebHelper.cs
namespace WorldFeed.Common.Helpers
{
    using System;
    using System.Collections.Generic;
    using System.IO;
    using System.Threading.Tasks;

    using WorldFeed.Common.Public.Models.Interfaces;

    public interface IWebHelper
    {
        Task<Stream> GetResponseStreamAsync(ITwitterRequest request);

        Dictionary<string, string> GetUriParameters(Uri uri);

        Dictionary<string, string> GetURLParameters(string url);

        Dictionary<string, string> GetQueryParameters(string queryUrl);

        string GetBaseURL(string url);

        string GetBaseURL(Uri uri);
    }
}

File: Sharebook/Client/src/app/core/Core/Helpers/IWebHelper.ts
import {inject, Inject, InjectionToken} from "@angular/core";
import {HttpClient} from "@angular/common/http";

import {ITwitterRequest} from "../../Public/Models/Interfaces/ITwitterRequest";
import {WebHelper} from "../../../webLogic/WebHelper";
import Uri from "typescript-dotnet-commonjs/System/Uri/Uri";
import Dictionary from "typescript-dotnet-commonjs/System/Collections/Dictionaries/Dictionary";

export interface IWebHelper {
  getResponseStreamAsync(request: ITwitterRequest): Promise<any>; // Task<Stream>;

  getUriParameters(uri: Uri): Dictionary<string, string>;

  getURLParameters(url: string): Dictionary<string, string>;

  getQueryParameters(queryUrl: string): Dictionary<string, string>;

  getBaseURL(url: string): string;

  getBaseURL(uri: Uri): string;
}

export const IWebHelperToken = new InjectionToken<IWebHelper>('IWebHelper', {
  providedIn: 'root',
  factory: () => new WebHelper(inject(HttpClient)),
});
